Fix infrequent case of not finding Wi-Fi adaptor on some computers and resulting crash

iglendd opened the pull request:
### What does this PR do?
- Fix root cause of the crash, missing check for non-nil adaptor even when error was nil (normal case)
- Original Adaptor enumeration `GetAdaptersInfo()` function did not show all adaptors on some computers. Moreover it is obsolete and work only for IP4. More modern replacement `GetAdaptersAddresses()` API did not help either unless ALL interfaces enumeration flag had been added and with the flag it requires ~70kb. Since IP address calculation is currently not needed, able to replace on direct interface finding by `GetIfEntry2()` API.

### Motivation
Fix crash

### Describe how you validated your changes
Manually tested on mine and problematic IT laptop where previously it caused the problem.

### Additional Notes

iglendd committed:
Fix infrequent case of not finding Wi-Fi adaptor on some computers and resulting crash

- Fix root cause of the crash, missing check for non-nil adaptor even when error was nil (normal case)
- Original Adaptor enumeration GetAdaptersInfo() function did not show all adaptors on some computers.
  Moreover it is obsolete and work only for IP4. More modern replacement GetAdaptersAddresses() API did not
  help either unless ALL interfaces enumeration flag had been added and with the flag it requires ~70kb.
  Since IP address calculation is currently not needed, able to replace on direct interface finding by
  GetIfEntry2() API.

iglendd committed:
Lint and release notes

jack0x2 reviewed (commented):
Do we have any tests for this? Other than the mock tests?

iglendd committed:
Apply suggestions from code review

Co-authored-by: Brett Blue <84536271+brett0000FF@users.noreply.github.com>

iglendd committed:
Update fix-wifi-check-crash-when-no-itf-matched-817810011963f983.yaml

iglendd committed:
Search interface by Luid instead by Guid when calling GetIfEntry2() API

While previous commit which had been using InterfaceGuid had been working
it looks like it was relying on the undocumented behavior which is addressed in this commit.

lrbhatia reviewed (approved):
Thanks! Overall, this looks good to me. I added one more comment to be addressed before merging this PR.

iglendd committed:
Code revie related, replaced partial struct on its full struct alternative (from windows package)

iglendd commented on pkg/collector/corechecks/net/wlan/wlan_windows.go:
```diff
@@ -225,96 +214,28 @@ type WLAN_CONNECTION_ATTRIBUTES struct {
+	PermanentPhysicalAddress [32]byte
+	Mtu                      uint32
+	Type                     uint32
+	// there are more fields below  but we do not need to access them
```
Spend some time to figureout full size filler but realized that windows package already have this struct. See the commit below. Thank you.

iglendd committed:
Merge branch 'main' into len.gamburg/fix-wifi-check-crash-when-no-itf-matched

iglendd commented:
/merge
